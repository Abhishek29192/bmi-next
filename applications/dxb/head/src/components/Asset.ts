import { graphql } from "gatsby";

export type Data = {
  resize: {
    src: string;
  };
  file: {
    fileName: string;
    url: string;
  };
};

export const query = graphql`
  fragment AssetFragment on ContentfulAsset {
    fluid {
      ...GatsbyContentfulFluid
    }
    resize(width: 1000, toFormat: JPG, jpegProgressive: true, quality: 60) {
      src
    }
    file {
      fileName
      url
    }
  }
`;
