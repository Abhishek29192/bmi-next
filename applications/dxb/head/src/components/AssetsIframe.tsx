import classnames from "classnames";
import React from "react";
import styles from "./styles/AssetsIframe.module.scss";

type Props = {
  url: string;
  title: string;
  className?: string;
};

const AssetsIframe = ({ url, title, className, ...rest }: Props) => (
  <iframe
    className={classnames(styles["assetsIframe"], className)}
    src={url}
    title={title}
    {...rest}
  />
);

export default AssetsIframe;
