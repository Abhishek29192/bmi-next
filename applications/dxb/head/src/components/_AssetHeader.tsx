import React, { useState } from "react";
import {
  AnchorLink,
  Button,
  Dialog,
  Icon,
  Tooltip
} from "@bmi-digital/components";
import ClickAwayListener from "@material-ui/core/ClickAwayListener";
import InfoIcon from "@material-ui/icons/Info";
import { microCopy } from "../constants/microCopies";
import RichText from "./RichText";
import { useSiteContext } from "./Site";
import styles from "./styles/DocumentTechnicalTableResults.module.scss";
import { PIMDocumentData, PIMLinkDocumentData } from "./types/PIMDocumentBase";

const AssetHeader = ({
  assetType
}: {
  assetType: PIMDocumentData["assetType"] | PIMLinkDocumentData["assetType"];
}) => {
  const { name, code, description } = assetType;
  const [isTooltipActive, setIsTooltipActive] = useState<boolean>(false);
  const [isDialogOpen, setIsDialogOpen] = useState<boolean>(false);
  const { getMicroCopy } = useSiteContext();

  return (
    <>
      {description && (
        <Dialog open={isDialogOpen} onCloseClick={() => setIsDialogOpen(false)}>
          <Dialog.Title>
            {name} ({code})
          </Dialog.Title>
          <Dialog.Content>
            <RichText document={description} />
          </Dialog.Content>
          <Dialog.Actions
            confirmLabel={getMicroCopy(microCopy.DIALOG_CLOSE)}
            onConfirmClick={() => setIsDialogOpen(false)}
          />
        </Dialog>
      )}
      <ClickAwayListener
        onClickAway={() => {
          setIsTooltipActive(false);
        }}
      >
        <Tooltip
          PopperProps={{
            disablePortal: true
          }}
          title={
            <>
              {name}
              {description && (
                <>
                  <br />
                  <AnchorLink
                    onClick={() => setIsDialogOpen(true)}
                    color="white"
                  >
                    {getMicroCopy(microCopy.GLOBAL_LEARN_MORE)}...
                  </AnchorLink>
                </>
              )}
            </>
          }
          open={isTooltipActive}
          interactive={!!description}
        >
          <div>
            <Button
              isIconButton
              variant="text"
              size={42}
              onClick={() =>
                setIsTooltipActive(
                  (prevIsTooltipActive) => !prevIsTooltipActive
                )
              }
            >
              <Icon source={InfoIcon} className={styles["tooltip-icon"]} />
            </Button>
          </div>
        </Tooltip>
      </ClickAwayListener>
    </>
  );
};

export default AssetHeader;
