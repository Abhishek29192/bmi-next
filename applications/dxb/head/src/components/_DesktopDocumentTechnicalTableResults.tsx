import React, { useContext } from "react";
import classnames from "classnames";
import {
  Button,
  Clickable,
  ClickableProps,
  DownloadList,
  DownloadListContext,
  Icon,
  IconButtonProps,
  iconMap,
  Table
} from "@bmi/components";
import axios from "axios";
import { useGoogleReCaptcha } from "react-google-recaptcha-v3";
import { downloadAs } from "../utils/client-download";
import withGTM from "../utils/google-tag-manager";
import { useConfig } from "../contexts/ConfigProvider";
import { microCopy } from "../constants/microCopies";
import AssetHeader from "./_AssetHeader";
import { useSiteContext } from "./Site";
import { Data as AssetTypeData } from "./AssetType";
import styles from "./styles/DocumentTechnicalTableResults.module.scss";
import { Format } from "./types";
import { PIMDocumentData, PIMLinkDocumentData } from "./types/PIMDocumentBase";
import createAssetFileCountMap, {
  AssetUniqueFileCountMap,
  generateFilenameByRealFileName,
  generateFileNamebyTitle
} from "./DocumentFileUtils";

interface Props {
  documentsByProduct: [string, (PIMDocumentData | PIMLinkDocumentData)[]][];
  assetTypes: AssetTypeData[];
  fileIconsMap: Record<Format, React.ComponentType>;
}

const DesktopDocumentTechnicalTableResults = ({
  documentsByProduct,
  assetTypes,
  fileIconsMap
}: Props) => {
  const {
    config: { documentDownloadEndpoint }
  } = useConfig();
  const { getMicroCopy } = useSiteContext();
  const { list } = useContext(DownloadListContext);
  const { executeRecaptcha } = useGoogleReCaptcha();

  const GTMClickable = withGTM<ClickableProps>(Clickable);
  const GTMButton = withGTM<IconButtonProps>(Button);

  const singleDocument = (asset: PIMDocumentData | PIMLinkDocumentData) =>
    asset.__typename !== "PIMLinkDocument" ? (
      <GTMClickable
        model="download"
        href={asset.url}
        download={asset.title}
        gtm={{
          id: "download1",
          label: "Download",
          action: asset.url
        }}
      >
        <Icon
          source={fileIconsMap[asset.format]}
          className={styles["format-icon"]}
        />
      </GTMClickable>
    ) : (
      <GTMButton
        isIconButton
        accessibilityLabel="Download"
        variant="text"
        action={{
          model: "htmlLink",
          href: asset.url,
          target: "_blank",
          rel: "noopener noreferrer"
        }}
        gtm={{
          id: "download1",
          label: "Download",
          action: asset.url
        }}
        className={styles["external-download-button"]}
        disableTouchRipple={true}
      >
        <Icon
          source={iconMap.External}
          className={styles["external-link-icon"]}
        />
      </GTMButton>
    );

  const multipleDocuments = (
    assets: (PIMDocumentData | PIMLinkDocumentData)[]
  ) => {
    const downloadMultipleFiles = async () => {
      try {
        if (!documentDownloadEndpoint) {
          throw Error(
            "`GATSBY_DOCUMENT_DOWNLOAD_ENDPOINT` missing in environment config"
          );
        }
        const [currentTime] = new Date()
          .toJSON()
          .replace(/-|:|T/g, "")
          .split(".");
        let zipFileName = `BMI_${currentTime}.zip`;
        if (assets.length > 0 && assets[0].product && assets[0].assetType) {
          zipFileName = `${assets[0].product?.name} ${assets[0].assetType.name}.zip`;
        }

        const token = await executeRecaptcha();
        const pimDocumentAssets: PIMDocumentData[] = assets.filter(
          (asset): asset is PIMDocumentData =>
            asset.__typename === "PIMDocument"
        );
        const assetFileCountMap: AssetUniqueFileCountMap =
          createAssetFileCountMap(pimDocumentAssets);
        const documents = pimDocumentAssets.map((asset, index) => ({
          href: asset.url,
          name:
            asset.realFileName && asset.realFileName !== ""
              ? generateFilenameByRealFileName(assetFileCountMap, asset, index)
              : generateFileNamebyTitle(
                  assetFileCountMap,
                  asset.title,
                  asset.extension,
                  index
                )
        }));
        const response = await axios.post(
          documentDownloadEndpoint,
          { documents: documents },
          { responseType: "text", headers: { "X-Recaptcha-Token": token } }
        );
        await downloadAs(response.data.url, zipFileName);
      } catch (error) {
        console.error("Download multiple documents", error); // eslint-disable-line
      }
    };

    return (
      <GTMButton
        isIconButton
        accessibilityLabel="Download"
        variant="text"
        action={{
          model: "default"
        }}
        onClick={downloadMultipleFiles}
        gtm={{
          id: "download1",
          label: "Download",
          action: JSON.stringify(
            Object.values(assets).map((asset) => asset.url)
          )
        }}
        disableTouchRipple={true}
        className={styles["external-download-button"]}
      >
        <Icon source={iconMap.FileZIP} className={styles["format-icon"]} />
      </GTMButton>
    );
  };

  return (
    <div className={styles["table-div"]}>
      <Table rowBgColorPattern="none">
        <Table.Head>
          <Table.Row className={styles["header-row"]}>
            <Table.Cell>
              {getMicroCopy(microCopy.DOCUMENT_LIBRARY_HEADERS_PRODUCT)}
            </Table.Cell>
            {assetTypes.map((assetType, index) => {
              return (
                <Table.Cell
                  key={`asset-type-${assetType.id}-${index}`}
                  className={styles["asset-type-cell"]}
                >
                  {assetType.code}
                  <AssetHeader assetType={assetType} />
                </Table.Cell>
              );
            })}
            <Table.Cell className={styles["all-files-header"]}>
              <span className={styles["all-files-header-wrapper"]}>
                <Icon
                  source={iconMap.Download}
                  className={styles["all-files-icon"]}
                />
                <span>
                  {getMicroCopy(microCopy.DOCUMENT_LIBRARY_HEADERS_ALL_FILES)}
                </span>
              </span>
            </Table.Cell>
          </Table.Row>
        </Table.Head>
        <Table.Body>
          {documentsByProduct.map(([productName, assets], index) => {
            const key = `${
              assets.find(({ product }) => product.code)?.product.code
            }-${index}`;
            const hasOnlyExternalAssets = assets.every(
              (asset) => asset.__typename === "PIMLinkDocument"
            );
            return (
              <Table.Row
                key={key}
                className={classnames(styles["row"], {
                  // eslint-disable-next-line security/detect-object-injection
                  [styles["row--checked"]]: !!list[key]
                })}
              >
                <Table.Cell>
                  {assets.length > 0 ? assets[0].product.name : productName}
                </Table.Cell>
                {assetTypes.map((assetType, index) => {
                  const filteredAssets = assets.filter(
                    ({ assetType: { id } }) => id === assetType.id
                  );

                  if (!filteredAssets.length) {
                    return (
                      <Table.Cell
                        key={`${productName}-missing-asset-${index}`}
                        className={styles["align-center"]}
                      >
                        <Icon
                          source={iconMap.Cross}
                          className={styles["no-document-icon"]}
                        />
                      </Table.Cell>
                    );
                  }

                  return (
                    <Table.Cell
                      key={`${productName}-asset-${assetType.id}`}
                      className={styles["align-center"]}
                    >
                      {filteredAssets.length === 1
                        ? singleDocument(filteredAssets[0])
                        : multipleDocuments(filteredAssets)}
                    </Table.Cell>
                  );
                })}
                <Table.Cell className={styles["align-center"]}>
                  {!hasOnlyExternalAssets && (
                    <DownloadList.Checkbox
                      name={key}
                      maxLimitReachedLabel={getMicroCopy(
                        microCopy.DOCUMENTS_DOWNLOAD_MAX_REACHED
                      )}
                      ariaLabel={`${getMicroCopy(
                        microCopy.DOCUMENT_LIBRARY_DOWNLOAD
                      )} ${productName}`}
                      value={assets.filter(
                        ({ __typename }) => __typename !== "PIMLinkDocument"
                      )}
                      fileSize={assets.reduce((acc, curr) => {
                        if (curr.__typename === "PIMLinkDocument") {
                          return 0;
                        }
                        return acc + (curr.fileSize || 0);
                      }, 0)}
                    />
                  )}
                </Table.Cell>
              </Table.Row>
            );
          })}
        </Table.Body>
      </Table>
    </div>
  );
};

export default DesktopDocumentTechnicalTableResults;
