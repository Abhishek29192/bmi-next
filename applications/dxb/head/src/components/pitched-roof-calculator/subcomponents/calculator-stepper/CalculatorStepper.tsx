import ArrowBackIcon from "@bmi-digital/components/icon/ArrowBack";
import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";
import classnames from "classnames";
import React, { FormEvent } from "react";
import ProgressIndicator from "../../../ProgressIndicator";
import {
  classes,
  FooterButton,
  Root,
  StepWrapper,
  StyledComputation,
  StyledContent,
  StyledFooter,
  StyledFooterBackButtonLink,
  StyledForm,
  StyledFormContent,
  StyledFormSubmitButton,
  StyledHR,
  StyledSkipAndNextButtons,
  StyledSpinnerContainer,
  StyledStepTitle,
  StyledSubTitle
} from "./CalculatorStepper.styles";

export type Props = {
  selected: string;
  children: React.ReactNode;
  loading: boolean;
};

const isStepElement = (
  element: React.ReactNode
): element is React.ReactElement<StepProps, typeof Step> =>
  !!element &&
  typeof element === "object" &&
  "type" in element &&
  element.type === Step;

const CalculatorStepper = ({ selected, children, loading }: Props) => {
  // TODO: add warnings for invalid items/keys
  const current = React.Children.toArray(children).filter((item) => {
    return isStepElement(item) && (item.key + "").substring(2) === selected;
  });

  return (
    <Root>
      {loading ? (
        <StyledSpinnerContainer>
          <ProgressIndicator size={40} />
        </StyledSpinnerContainer>
      ) : (
        current
      )}
    </Root>
  );
};

export type StepProps = {
  isForm?: boolean;
  title: string;
  subtitle: React.ReactNode;
  paragraph?: React.ReactNode;
  backLabel?: string;
  backButtonOnClick?: () => void;
  linkLabel?: string;
  linkOnClick?: () => void;
  nextLabel?: string;
  nextButtonOnClick?: (e: FormEvent, value: Record<string, unknown>) => void;
  children: React.ReactNode;
};

const Step = ({
  isForm = true,
  title,
  subtitle,
  paragraph,
  backLabel,
  backButtonOnClick,
  linkLabel,
  linkOnClick,
  nextLabel,
  nextButtonOnClick,
  children
}: StepProps) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("lg"));

  const backButton = backLabel ? (
    <FooterButton
      variant="outlined"
      onClick={backButtonOnClick}
      accessibilityLabel="Back"
      startIcon={<ArrowBackIcon />}
    >
      {backLabel}
    </FooterButton>
  ) : null;

  const link = linkLabel ? (
    <StyledFooterBackButtonLink onClick={linkOnClick} variant="text">
      {linkLabel}
    </StyledFooterBackButtonLink>
  ) : null;

  const nextButton = nextLabel ? (
    <StyledFormSubmitButton accessibilityLabel="Next" component={FooterButton}>
      {nextLabel}
    </StyledFormSubmitButton>
  ) : null;

  const totalFooterButtons =
    Number(Boolean(nextButton)) +
    Number(Boolean(backButton)) +
    Number(Boolean(link));

  const footer = totalFooterButtons ? (
    <StyledFooter>
      {isMobile ? (
        <>
          {nextButton}
          {backButton}
          {link}
        </>
      ) : (
        <>
          {backButton}
          <StyledSkipAndNextButtons>
            {link}
            {nextButton}
          </StyledSkipAndNextButtons>
        </>
      )}
    </StyledFooter>
  ) : null;

  const content = (
    <>
      <StyledFormContent className={classes["formContent"]}>
        <StyledStepTitle variant="h4">{title}</StyledStepTitle>
        {paragraph && (
          <StyledComputation variant="h6">{paragraph}</StyledComputation>
        )}
        {subtitle && (
          <StyledSubTitle variant="body1" className={classes["subtitle"]}>
            {subtitle}
          </StyledSubTitle>
        )}
        <StyledContent className={classes["content"]}>{children}</StyledContent>
        {!footer ? <StyledHR /> : null}
      </StyledFormContent>
      {footer ? footer : null}
    </>
  );

  const paddingClasses = classnames(
    totalFooterButtons === 1 && classes["form-sm-padding"],
    totalFooterButtons === 2 && classes["form-md-padding"],
    totalFooterButtons === 3 && classes["form-lg-padding"]
  );

  return isForm ? (
    <StepWrapper>
      <StyledForm
        data-testid="calculator-step-form"
        onSubmit={nextButtonOnClick}
        className={paddingClasses}
      >
        {content}
      </StyledForm>
    </StepWrapper>
  ) : (
    <StepWrapper>
      <div className={paddingClasses}>{content}</div>
    </StepWrapper>
  );
};

CalculatorStepper.Step = Step;

export default CalculatorStepper;
