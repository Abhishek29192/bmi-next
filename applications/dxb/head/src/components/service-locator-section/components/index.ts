// istanbul ignore file: doesn't hold any logic
export * from "./CompanyDetails";
export * from "./IntegratedLinkCard";
export * from "./ServiceLocationBlock/SearchLocationBlock";
export * from "./ServiceLocatorChips/ServiceLocatorChips";
export * from "./ServiceLocatorMap";
export * from "./ServiceLocatorResultList/ServiceLocatorResultList";
export * from "./useScrollTo";
