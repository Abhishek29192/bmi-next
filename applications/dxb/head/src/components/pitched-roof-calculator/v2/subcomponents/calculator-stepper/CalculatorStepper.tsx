import { Button, Form, Typography } from "@bmi/components";
import { useMediaQuery, useTheme } from "@material-ui/core";
import CircularProgress from "@material-ui/core/CircularProgress";
import Divider from "@material-ui/core/Divider";
import ArrowBackIcon from "@material-ui/icons/ArrowBack";
import classnames from "classnames";
import React, { FormEvent } from "react";
import styles from "./CalculatorStepper.module.scss";

export type Props = {
  selected: string;
  children: React.ReactNode;
  loading: boolean;
};

const isStepElement = (
  element: React.ReactNode
): element is React.ReactElement<StepProps, typeof Step> =>
  !!element &&
  typeof element === "object" &&
  "type" in element &&
  element.type === Step;

const CalculatorStepper = ({ selected, children, loading }: Props) => {
  // TODO: add warnings for invalid items/keys
  const current = React.Children.toArray(children).filter((item) => {
    return isStepElement(item) && (item.key + "").substring(2) === selected;
  });

  return (
    <div className={styles["CalculatorStepper"]}>
      {loading ? (
        <div className={styles["spinnerContainer"]}>
          <CircularProgress className={styles["spinner"]} />
        </div>
      ) : (
        current
      )}
    </div>
  );
};

export type StepProps = {
  isForm?: boolean;
  title: string;
  subtitle: string;
  paragraph?: React.ReactNode;
  backLabel?: string;
  backButtonOnClick?: () => void;
  linkLabel?: string;
  linkOnClick?: () => void;
  nextLabel?: string;
  nextButtonOnClick?: (e: FormEvent, value: Record<string, unknown>) => void;
  children: React.ReactNode;
};

const Step = ({
  isForm = true,
  title,
  subtitle,
  paragraph,
  backLabel,
  backButtonOnClick,
  linkLabel,
  linkOnClick,
  nextLabel,
  nextButtonOnClick,
  children
}: StepProps) => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down("sm"));

  const backButton = backLabel ? (
    <Button
      variant="outlined"
      onClick={backButtonOnClick}
      accessibilityLabel="Back"
      startIcon={<ArrowBackIcon />}
      className={styles["footerButton"]}
    >
      {backLabel}
    </Button>
  ) : null;

  const link = linkLabel ? (
    <Button
      onClick={linkOnClick}
      variant="text"
      className={classnames(
        styles["footerButton"],
        styles["footerButton--link"]
      )}
    >
      {linkLabel}
    </Button>
  ) : null;

  const nextButton = nextLabel ? (
    <Form.SubmitButton
      className={styles["footerButton"]}
      accessibilityLabel="Next"
    >
      {nextLabel}
    </Form.SubmitButton>
  ) : null;

  const totalFooterButtons =
    Number(Boolean(nextButton)) +
    Number(Boolean(backButton)) +
    Number(Boolean(link));

  const footer = totalFooterButtons ? (
    <div className={classnames(styles["footer"])}>
      {isMobile ? (
        <>
          {nextButton}
          {backButton}
          {link}
        </>
      ) : (
        <>
          {backButton}
          <div className={styles["skipAndNextButtons"]}>
            {link}
            {nextButton}
          </div>
        </>
      )}
    </div>
  ) : null;

  const content = (
    <>
      <div className={styles["formContent"]}>
        <Typography variant="h4" className={classnames(styles["step-title"])}>
          {title}
        </Typography>
        <Typography variant="h6" className={classnames(styles["computation"])}>
          {paragraph}
        </Typography>
        <Typography variant="body1" className={classnames(styles["subtitle"])}>
          {subtitle}
        </Typography>
        <div className={styles["content"]}>{children}</div>
        {!footer ? <Divider className={styles["hr"]} /> : null}
      </div>
      {footer ? footer : null}
    </>
  );

  const paddingClasses = classnames(
    totalFooterButtons === 1 && styles["form--sm-padding"],
    totalFooterButtons === 2 && styles["form--md-padding"],
    totalFooterButtons === 3 && styles["form--lg-padding"]
  );

  return isForm ? (
    <Form
      data-testid="calculator-step-form"
      onSubmit={nextButtonOnClick}
      className={classnames(paddingClasses, styles["form"])}
    >
      {content}
    </Form>
  ) : (
    <div className={classnames(paddingClasses, styles["stepWithoutForm"])}>
      {content}
    </div>
  );
};

CalculatorStepper.Step = Step;

export default CalculatorStepper;
