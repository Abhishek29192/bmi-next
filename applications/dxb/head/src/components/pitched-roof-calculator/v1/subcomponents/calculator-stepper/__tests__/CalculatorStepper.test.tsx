import { ThemeProvider } from "@bmi-digital/components";
import { fireEvent, render } from "@testing-library/react";
import mockConsole from "jest-mock-console";
import React from "react";
import CalculatorStepper from "../CalculatorStepper";

beforeAll(() => {
  mockConsole();
});

describe("CalculatorStepper component", () => {
  const SelectingARoof = () => {
    return (
      <div>
        <p>This is the Selecting a Roof Shape page.</p>
      </div>
    );
  };

  it("renders correctly", () => {
    const { container } = render(
      <ThemeProvider>
        <CalculatorStepper selected="select-a-roof-shape">
          <CalculatorStepper.Step
            key="select-a-roof-shape"
            title="This is the 2nd page"
            subtitle="Choose the closest to your roof shape"
            backLabel="Go Back"
            backButtonOnClick={() => {}}
            linkLabel="Skip"
            linkOnClick={() => {}}
            nextLabel="Calculate"
            nextButtonOnClick={() => {}}
          >
            <SelectingARoof />
          </CalculatorStepper.Step>
        </CalculatorStepper>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("renders as a div", () => {
    const { container } = render(
      <ThemeProvider>
        <CalculatorStepper selected="select-a-roof-shape">
          <CalculatorStepper.Step
            isForm={false}
            key="select-a-roof-shape"
            title="This is the 2nd page"
            subtitle="Choose the closest to your roof shape"
            backLabel="Go Back"
            backButtonOnClick={() => {}}
            linkLabel="Skip"
            linkOnClick={() => {}}
            nextLabel="Calculate"
            nextButtonOnClick={() => {}}
          >
            <SelectingARoof />
          </CalculatorStepper.Step>
        </CalculatorStepper>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("ignores footer when there are no buttons", () => {
    const { container } = render(
      <ThemeProvider>
        <CalculatorStepper selected="select-a-roof-shape">
          <CalculatorStepper.Step
            key="select-a-roof-shape"
            title="This is the 2nd page"
            subtitle="Choose the closest to your roof shape"
          >
            <SelectingARoof />
          </CalculatorStepper.Step>
        </CalculatorStepper>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("calls button events", () => {
    const onClickBack = jest.fn();
    const backLabel = "Go Back";

    const onClickLink = jest.fn();
    const linkLabel = "Skip";

    const onClickNext = jest.fn();
    const nextLabel = "Calculate";

    const { getByText } = render(
      <ThemeProvider>
        <CalculatorStepper selected="select-a-roof-shape">
          <CalculatorStepper.Step
            key="select-a-roof-shape"
            title="This is the 2nd page"
            subtitle="Choose the closest to your roof shape"
            backLabel={backLabel}
            backButtonOnClick={onClickBack}
            linkLabel={linkLabel}
            linkOnClick={onClickLink}
            nextLabel={nextLabel}
            nextButtonOnClick={onClickNext}
          >
            <SelectingARoof />
          </CalculatorStepper.Step>
        </CalculatorStepper>
      </ThemeProvider>
    );

    const backButton = getByText(backLabel);
    fireEvent.click(backButton);
    expect(onClickBack.mock.calls).toMatchSnapshot();

    const linkButton = getByText(linkLabel);
    fireEvent.click(linkButton);
    expect(onClickLink.mock.calls).toMatchSnapshot();

    const nextButton = getByText(nextLabel);
    fireEvent.click(nextButton);
    expect(onClickNext.mock.calls).toMatchSnapshot();
  });
});
