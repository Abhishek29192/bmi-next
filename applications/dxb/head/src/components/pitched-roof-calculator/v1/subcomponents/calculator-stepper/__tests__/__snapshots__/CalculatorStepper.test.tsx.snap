// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorStepper component calls button events 1`] = `
Array [
  Array [
    Class {
      "_dispatchInstances": null,
      "_dispatchListeners": null,
      "_targetInst": null,
      "altKey": null,
      "bubbles": null,
      "button": null,
      "buttons": null,
      "cancelable": null,
      "clientX": null,
      "clientY": null,
      "ctrlKey": null,
      "currentTarget": [Function],
      "defaultPrevented": null,
      "detail": null,
      "dispatchConfig": null,
      "eventPhase": null,
      "getModifierState": [Function],
      "isDefaultPrevented": [Function],
      "isPropagationStopped": [Function],
      "isTrusted": null,
      "metaKey": null,
      "movementX": [Function],
      "movementY": [Function],
      "nativeEvent": null,
      "pageX": null,
      "pageY": null,
      "relatedTarget": [Function],
      "screenX": null,
      "screenY": null,
      "shiftKey": null,
      "target": null,
      "timeStamp": [Function],
      "type": null,
      "view": null,
    },
  ],
]
`;

exports[`CalculatorStepper component calls button events 2`] = `
Array [
  Array [
    Class {
      "_dispatchInstances": null,
      "_dispatchListeners": null,
      "_targetInst": null,
      "altKey": null,
      "bubbles": null,
      "button": null,
      "buttons": null,
      "cancelable": null,
      "clientX": null,
      "clientY": null,
      "ctrlKey": null,
      "currentTarget": [Function],
      "defaultPrevented": null,
      "detail": null,
      "dispatchConfig": null,
      "eventPhase": null,
      "getModifierState": [Function],
      "isDefaultPrevented": [Function],
      "isPropagationStopped": [Function],
      "isTrusted": null,
      "metaKey": null,
      "movementX": [Function],
      "movementY": [Function],
      "nativeEvent": null,
      "pageX": null,
      "pageY": null,
      "relatedTarget": [Function],
      "screenX": null,
      "screenY": null,
      "shiftKey": null,
      "target": null,
      "timeStamp": [Function],
      "type": null,
      "view": null,
    },
  ],
]
`;

exports[`CalculatorStepper component calls button events 3`] = `
Array [
  Array [
    SyntheticEvent {
      "_dispatchInstances": null,
      "_dispatchListeners": null,
      "_targetInst": null,
      "bubbles": null,
      "cancelable": null,
      "currentTarget": [Function],
      "defaultPrevented": null,
      "dispatchConfig": null,
      "eventPhase": null,
      "isDefaultPrevented": [Function],
      "isPropagationStopped": [Function],
      "isTrusted": null,
      "nativeEvent": null,
      "target": null,
      "timeStamp": [Function],
      "type": null,
    },
    Object {},
  ],
]
`;

exports[`CalculatorStepper component ignores footer when there are no buttons 1`] = `
<div>
  <div
    class="CalculatorStepper"
  >
    <form
      class="Form-root-67"
    >
      <h4
        class="MuiTypography-root step-title Typography-root-72 MuiTypography-h4"
        data-testid="typography"
      >
        This is the 2nd page
      </h4>
      <h6
        class="MuiTypography-root computation Typography-root-72 MuiTypography-h6"
        data-testid="typography"
      />
      <p
        class="MuiTypography-root subtitle Typography-root-72 MuiTypography-body1"
        data-testid="typography"
      >
        Choose the closest to your roof shape
      </p>
      <div
        class="content"
      >
        <div>
          <p>
            This is the Selecting a Roof Shape page.
          </p>
        </div>
      </div>
      <hr
        class="MuiDivider-root hr"
      />
    </form>
  </div>
</div>
`;

exports[`CalculatorStepper component renders as a div 1`] = `
<div>
  <div
    class="CalculatorStepper"
  >
    <div>
      <h4
        class="MuiTypography-root step-title Typography-root-34 MuiTypography-h4"
        data-testid="typography"
      >
        This is the 2nd page
      </h4>
      <h6
        class="MuiTypography-root computation Typography-root-34 MuiTypography-h6"
        data-testid="typography"
      />
      <p
        class="MuiTypography-root subtitle Typography-root-34 MuiTypography-body1"
        data-testid="typography"
      >
        Choose the closest to your roof shape
      </p>
      <div
        class="content"
      >
        <div>
          <p>
            This is the Selecting a Roof Shape page.
          </p>
        </div>
      </div>
      <hr
        class="MuiDivider-root hr"
      />
      <div
        class="footer breakableRow breakableRow--reverse"
      >
        <div
          class="breakableRow"
        >
          <button
            class="MuiButtonBase-root MuiButton-root Button-root-42 MuiButton-outlined rowItem MuiButton-outlinedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label Button-label-45"
            >
              <span
                class="MuiButton-startIcon Button-startIcon-46 MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
              Go Back
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="breakableRow"
        >
          <button
            class="MuiButtonBase-root MuiButton-root Button-root-42 MuiButton-text Button-text-43 rowItem MuiButton-textPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label Button-label-45"
            >
              Skip
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root Button-root-42 MuiButton-contained Button-contained-44 rowItem MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label Button-label-45"
            >
              Calculate
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CalculatorStepper component renders correctly 1`] = `
<div>
  <div
    class="CalculatorStepper"
  >
    <form
      class="Form-root-1"
    >
      <h4
        class="MuiTypography-root step-title Typography-root-6 MuiTypography-h4"
        data-testid="typography"
      >
        This is the 2nd page
      </h4>
      <h6
        class="MuiTypography-root computation Typography-root-6 MuiTypography-h6"
        data-testid="typography"
      />
      <p
        class="MuiTypography-root subtitle Typography-root-6 MuiTypography-body1"
        data-testid="typography"
      >
        Choose the closest to your roof shape
      </p>
      <div
        class="content"
      >
        <div>
          <p>
            This is the Selecting a Roof Shape page.
          </p>
        </div>
      </div>
      <hr
        class="MuiDivider-root hr"
      />
      <div
        class="footer breakableRow breakableRow--reverse"
      >
        <div
          class="breakableRow"
        >
          <button
            class="MuiButtonBase-root MuiButton-root Button-root-14 MuiButton-outlined rowItem MuiButton-outlinedPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label Button-label-17"
            >
              <span
                class="MuiButton-startIcon Button-startIcon-18 MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
                  />
                </svg>
              </span>
              Go Back
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="breakableRow"
        >
          <button
            class="MuiButtonBase-root MuiButton-root Button-root-14 MuiButton-text Button-text-15 rowItem MuiButton-textPrimary"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label Button-label-17"
            >
              Skip
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root Button-root-14 MuiButton-contained Button-contained-16 rowItem MuiButton-containedPrimary"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label Button-label-17"
            >
              Calculate
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;
